enhancements/cluster-logging/forwarding-json-structured-logs.md

```diff
@@ -128,154 +101,107 @@ New *pipeline* field:
 For each log entry; if `parse: json` is set _and_ the entry is valid JSON, the output record will include a `structured` field _equivalent_ to the JSON entry.
 It may differ in field order and use of white-space.
 
-All record will have a `message` field for backwards compatibility.
+### Output type elasticsearch
 
-#### Output  configuration for type:elasticsearch
+For most output types it is sufficient to enable `parse: json` to forward JSON data.
 
-New fields for output type `elasticsearch`
-* `indexKey`: (string, optional) Use value of meta-data key as `index` value, if present.
-  These keys are supported:
-  - `kubernetes.namespaceName`: Use the namespace name as the index name.
-  - `kubernetes.labels.<key>`: Use the string value of kubernetes label with key `<key>`.
-  - `openshift.labels.<key>`: Use the string value of an openshift label with key `<key>` (see [](forwarder-tagging.md))
-  - *other keys may be added in future*
-* `indexName`: (string, optional)
-  * If `indexKey` is not set, or the key is missing, use `indexName` as the `index` value.
-  * If `indexKey` is set and the key is present, that takes precedence over `indexName`
+Elasticsearch is a special case; JSON records with _different formats_ must go to different indices, otherwise type conflicts and cardinality problems can occur.
 
-**Note:** The Elasticsearch output will _delete_ the `structured` field if the rules above do not yield a non-empty index name.
-This is to avoid sending randomly structured JSON to shared indices.
+**Note**: It is important not to overload elasticsearch with too many indices.
+Create new indices for each different log _format_, **not** for each application or namespace.
+For example, most programs from Apache log with the same JSON format.
+All such logs should go to one index, even if they come from different namespaces or applications.
+Once stored, Elasticsearch queries can separate logs by application, namespace and many other criteria.
 
-Indices are created automatically on-demand by co-operation between the forwarder and the managed default store.
+The following new sub-fields in the `elasticsearch` output field allow JSON records to be directed to different indices.
 
-#### Default output configuration
+**Note**: The index name specified below is not the complete index name, it will be
+prefixed with "app-" to match the indexing scheme of default elasticsearch.
```

Comment:
maybe we should explicitly mention it will be both prefixed, and aliased with `app`

Reply:
Done, but please clarify if we need to say more about the aliasing etc. to make this clear.